lib/teaching.g

```diff
@@ -742,8 +742,7 @@ return function(o,n)
   if p<>fail then
     if n=fail then
       # delete
-      OBJLIST[p]:=OBJLIST[Length(OBJLIST)];
```

Comment:
Like a previous case, this will fail to remove 'p' if it was the last element in OBJLIST.

Reply:
True. But in this case I think I'll just do the "obvious" fix and write `Remove(OBJLIST,p)`, which may be in theory a minimal deoptimization but this is teaching code so I think clearer is better.